src/operator/tensor/cast_storage-inl.cuh

```diff
@@ -188,6 +188,15 @@ struct FillRspValsKernel {
   }
 };
 
+template<typename xpu, int ndim, typename DType>
+inline mshadow::Tensor<xpu, ndim, DType> AllocateTempDataForCast(const OpContext& op_ctx,
```

Comment:
'inline' asks the compiler to in-line the function.  It is not just a linkage directive.